### DISCUSSION TITLE
https://theme-hope.vuejs.press/zh/guide/feature/comment.html

### DISCUSSION THREAD

**Discussion by giscus** [vuepress-theme-hope/giscus-discussions#527] [Announcements]
# https://theme-hope.vuejs.press/zh/guide/feature/comment.html

通过内置 ，vuepress-theme-hope 实现了评论功能。 相关信息 vuepress-theme-hope 将主题选项中的 plugins.comment 作为插件选项提供给 vuepress-plugin-comment2。

https://theme-hope.vuejs.press/zh/guide/feature/comment.html

**Comment by 870138612** [comment]
了解

**Comment by Nick-DL** [comment]
我这里waline评论相关代码如果写在config.ts里，navbar栏会变空白，但如果写在theme.ts里替换掉模板里的giscus，就没有问题

**Comment by LuTiantian-0406** [comment]
为什么giscus每次点评论一直都在加载呢

**Reply by Mister-Hope** [reply]
截图？或者复线步骤？如果你访问GitHub或者giscuss服务器不稳定，你可能会遇到加载上的问题

**Comment by weiguoli369** [comment]
使用waline，本地或者githubio上都有跨域访问问题，请问这个是要在哪添加运行跨域的参数？

**Reply by Kiritoeo** [reply]
```javascript
const Waline = require('@waline/vercel');

// 创建符合 ThinkJS 规范的中间件函数
const createWalineMiddleware = Waline({
  plugins: [],
  async postSave(comment) {
    // 原有逻辑
  },
});

// 包装中间件以添加 CORS 头
module.exports = async (req, res) => {
  // 设置 CORS 头
  res.setHeader('Access-Control-Allow-Origin', 'https://kiritoeo.github.io');
  res.setHeader('Access-Control-Allow-Methods', 'GET, POST, OPTIONS');
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type, Authorization');

  // 处理 OPTIONS 预检请求
  if (req.method === 'OPTIONS') {
    return res.status(200).end();
  }

  // 调用 Waline 中间件
  return createWalineMiddleware(req, res);
};
```
修改vercel生成的github项目里面的index.cjs文件

**Comment by zzysnbb** [comment]
[图片]

**Reply by zzysnbb** [reply]
[https://myoctocat.com/assets/images/base-octocat.svg]

**Reply by zzysnbb** [reply]
![Screenshot of a comment on a GitHub issue showing an image, added in the Markdown, of an Octocat smiling and raising a tentacle.](https://myoctocat.com/assets/images/base-octocat.svg)
